sdks/java/extensions/zetasketch/src/main/java/org/apache/beam/sdk/extensions/zetasketch/HllCount.java

```diff
@@ -0,0 +1,345 @@
+    }
+
+    /**
+     * Returns a {@link HllCount.Init.Builder<byte[]>} for a {@code HllCount.Init} combining {@code
+     * PTransform}. Call {@link Builder#globally()} or {@link Builder#perKey()} on the returning
+     * {@link Builder} to finalize the {@code PTransform}.
+     *
+     * <p>Calling {@link Builder#globally()} returns a {@code PTransform} that takes an input {@code
+     * PCollection<byte[]>} and returns a {@code PCollection<byte[]>} whose contents is the
+     * bytes-type HLL++ sketch computed from the elements in the input {@code PCollection}.
+     *
+     * <p>Calling {@link Builder#perKey()} returns a {@code PTransform} that takes an input {@code
+     * PCollection<KV<K, byte[]>>} and returns a {@code PCollection<KV<K, byte[]>>} whose contents
+     * is the per-key bytes-type HLL++ sketch computed from the values matching each key in the
+     * input {@code PCollection}.
+     *
+     * <p>Bytes-type sketches cannot be merged with sketches of other types.
+     */
+    public static Builder<byte[]> bytesSketch() {
+      return new Builder<>(HllCountInitFn.forBytes());
+    }
+
+    /**
+     * Builder for the {@code HllCount.Init} combining {@code PTransform}.
+     *
+     * <p>Call {@link #withPrecision(int)} to customize the {@code precision} parameter of the
+     * sketch.
+     *
+     * <p>Call {@link #globally()} or {@link #perKey()} to finalize the {@code PTransform}.
+     *
+     * @param <InputT> element type or value type in {@code KV}s of the input {@code PCollection} to
+     *     the {@code PTransform} being built
+     */
+    public static final class Builder<InputT> {
+
+      private final HllCountInitFn<InputT, ?> initFn;
+
+      private Builder(HllCountInitFn<InputT, ?> initFn) {
+        this.initFn = initFn;
+      }
+
+      /**
+       * Explicitly set the {@code precision} parameter used to compute HLL++ sketch.
+       *
+       * <p>Valid range is between {@value #MINIMUM_PRECISION} and {@value #MAXIMUM_PRECISION}. If
+       * this method is not called, {@value #DEFAULT_PRECISION} will be used. Sketches computed
+       * using different {@code precision}s cannot be merged together.
+       *
+       * @param precision the {@code precision} parameter used to compute HLL++ sketch
+       */
+      public Builder<InputT> withPrecision(int precision) {
+        initFn.setPrecision(precision);
+        return this;
+      }
+
+      /**
+       * Returns a {@link Combine.Globally} {@code PTransform} that takes an input {@code
+       * PCollection<InputT>} and returns a {@code PCollection<byte[]>} whose contents is the HLL++
+       * sketch computed from the elements in the input {@code PCollection}.
+       *
+       * <p>Returns an empty output {@code PCollection} if the input {@code PCollection} is empty.
+       */
+      public Combine.Globally<InputT, byte[]> globally() {
+        return Combine.globally(initFn).withoutDefaults();
```

Comment:
If HllCount cannot work with portable worker, consider checking experiment `beam_fn_api` and throws not support error?

Reply:
I run a test against the portable Dataflow worker, and unfortunately it failed to work due to some coder-related issues. I filed https://issues.apache.org/jira/browse/BEAM-7879 to track it. 

However, checking if the runner is portable runner at this SDK feature level is hard, because from the implementation of the `PTransform`s there is no reference to the pipeline options, and most transforms provided in this class are only wrappers around `Combine.Globally` or `Combine.PerKey`. So instead, I added a note and a link to the Jira ticket in the comment.

Reply:
When running `transform.expand()`, you can get `PipelineOption` from pipeline by calling `PCollection.getPipeline()`.

Reply:
`HllCount.Init` and `HllCount.MergePartial` here are not `PTransform`s themselves. They are APIs on top of a `Combine.Globally` or `Combine.PerKey` transform. So I don't believe I can have the check in the way you mentioned above.